controllers/scopewatcher_controller.go

```diff
@@ -98,6 +99,93 @@ func (r *CommonServiceReconciler) ScopeReconcile(ctx context.Context, req ctrl.R
 	// 6. Migrate Cert-Manager
 
 	// 7. Delete Crossplane, webhook, and secretshare deployment
+	var CP2Deployments = []*bootstrap.Resource{
+		{
+			Name:    "secretshare",
+			Version: "v1",
+			Group:   "apps",
+			Kind:    "Deployment",
+			Scope:   "namespaceScope",
+		},
+		{
+			Name:    "ibm-common-service-webhook",
+			Version: "v1",
+			Group:   "apps",
+			Kind:    "Deployment",
+			Scope:   "namespaceScope",
+		},
+	}
+
+	var CP2Resources = []*bootstrap.Resource{
+		{
+			Name:    "ibm-common-service-webhook",
+			Version: "v1alpha1",
+			Group:   "operator.ibm.com",
+			Kind:    "PodPreset",
+			Scope:   "namespaceScope",
+		},
+		{
+			Name:    "ibm-common-service-webhook-configuration",
+			Version: "v1",
+			Group:   "admissionregistration.k8s.io",
+			Kind:    "MutatingWebhookConfiguration",
+			Scope:   "clusterScope",
+		},
+		{
+			Name:    "ibm-operandrequest-webhook-configuration",
+			Version: "v1",
+			Group:   "admissionregistration.k8s.io",
+			Kind:    "MutatingWebhookConfiguration",
+			Scope:   "clusterScope",
+		},
+		{
+			Name:    "ibm-cs-ns-mapping-webhook-configuration",
+			Version: "v1",
+			Group:   "admissionregistration.k8s.io",
+			Kind:    "ValidatingWebhookConfiguration",
+			Scope:   "clusterScope",
+		},
+		{
+			Name:    "common-services",
+			Version: "v1",
+			Group:   "ibmcpcs.ibm.com",
+			Kind:    "SecretShare",
+			Scope:   "namespaceScope",
+		},
+	}
+
+	// remove crossplane
+	if err := r.Bootstrap.DeleteSubscription(constant.ICPPKOperator, r.Bootstrap.CSData.ControlNs); err != nil {
+		klog.Errorf("Failed to delete %s in %s: %v", constant.ICPPKOperator, r.Bootstrap.CSData.ControlNs, err)
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Bootstrap.DeleteSubscription(constant.ICPPICOperator, r.Bootstrap.CSData.ControlNs); err != nil {
+		klog.Errorf("Failed to delete %s in %s: %v", constant.ICPPICOperator, r.Bootstrap.CSData.ControlNs, err)
+		return ctrl.Result{}, err
+	}
+
+	if err := r.Bootstrap.DeleteSubscription(constant.ICPOperator, r.Bootstrap.CSData.ControlNs); err != nil {
+		klog.Errorf("Failed to delete %s in %s: %v", constant.ICPOperator, r.Bootstrap.CSData.ControlNs, err)
+		return ctrl.Result{}, err
+	}
+
+	if updateErr := r.Bootstrap.CreateorUpdateCFCrossplaneConfigMap("'true'"); updateErr != nil {
+		return ctrl.Result{}, updateErr
+	}
+
+	// remove webhook and secretshare
+	for _, deployment := range CP2Deployments {
+		if err := r.Bootstrap.Cleanup(r.Bootstrap.CSData.ControlNs, deployment); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+
+	for _, resource := range CP2Resources {
+		if err := r.Bootstrap.Cleanup(r.Bootstrap.CSData.ControlNs, resource); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
```

Comment:
Hi @qpdpQ 

Could we update the namespace for cleaning cluster singleton service to `MasterNs` which is typically `ibm-common-services`?
The main purpose of this cleaning is to migrate cluster singleton service from `MasterNs` to `ControlNs`. In another word, if it is in `ControlNs` already, we do not have to delete them.

Reply:
Also I am wondering that does the sequence of removing operator and operand matter?

Would we consider removing operand first, so operator will have chance to reconcile the resources, and remove the finalizers in the CRs if it is necessary.